lib/puppeteer/events.rb

```diff
@@ -0,0 +1,152 @@
+require 'digest/md5'
+
+module EventsDefinitionUtils
+  refine Kernel do
+    # Symbol is used to prevent external parties listening to these events
+    def Symbol(str)
+      Digest::MD5.hexdigest(str)
+    end
+  end
+
+  refine Hash do
+    def define_const_into(target_module)
+      self.each do |key, value|
+        target_module.const_set(key, value)
+        target_module.define_singleton_method(key) { value }
+      end
+      keyset = Set.new(self.keys)
+      valueset = Set.new(self.values)
+      target_module.define_singleton_method(:keys) { keyset }
+      target_module.define_singleton_method(:values) { valueset }
+    end
+  end
+end
+
+using EventsDefinitionUtils
+
+module Events ; end
+
+# @see https://github.com/puppeteer/puppeteer/blob/main/src/common/Events.ts
+{
+  Page: {
```

Comment:
It seems `Events.ts` will be removed in future.

It would be better to define these constants in PageEmittedEvents